src/helpers/validators/validateString/index.ts

```diff
@@ -0,0 +1,121 @@
+// import helpers
+import validateRegex from '../shared/helpers/validateRegex';
+import ValidationResult from '../shared/types/ValidationResult';
+
+// import types
+import StringValidationRequirements from './StringValidationRequirements';
+
+/**
+ * Determines whether a given input string is considered valid based on
+ *   the provided requirements.
+ * @author Austen Money
+ * @param input input string
+ * @param reqs requirements that the provided string should conform to
+ * @returns whether input is considered valid according to reqs - if
+ *   valid, returns a cleaned version of input; if invalid, returns
+ *   a string containing error messages describing which requirements
+ *   were not met.
+ */
+const validateString = (
+  input: string,
+  reqs: StringValidationRequirements,
+): ValidationResult => {
+  // stores all invalid input errors
+  const errorMessages: Array<string> = [];
+  // contains version of input that will be returned
+  let cleanedValue: string = input;
+
+  // remove whitespace if required
+  if (reqs.ignoreWhitespace) {
+    cleanedValue = input.replace(/\s+/g, '');
+  }
+
+  // apply max char requirement
+  if (reqs.minLen) {
+    if (cleanedValue.length < reqs.minLen) {
+      errorMessages.push(`Input must not be under ${reqs.minLen} character(s).`);
+    }
+  }
+
+  // apply max char requirement
+  if (reqs.maxLen) {
+    if (cleanedValue.length > reqs.maxLen) {
+      errorMessages.push(`Input must not be over ${reqs.maxLen} character(s).`);
+    }
+  }
+
+  // apply alphabetical requirement
+  if (reqs.lettersOnly) {
```

Comment:
This will throw errors for non-English letters, like á, ó, ö, and so on. If you want to accept any alphabetic character, then we'll be [going down a whole Unicode rabbit hole](https://stackoverflow.com/a/37511463), but I think it's worth mentioning.

Reply:
That's a very good point - I'm not very familiar with Unicode, but just went down the rabbit hole a bit and changed the `lettersOnly` requirement to remove diacritics before checking.  

I wonder if setting the expectation of "only letters" would imply letters within the ASCII range? If that's the case, would it make sense to return the string sans-diacritics?

Reply:
I want to make sure this is an inclusive script, so perhaps one way we could do this in the future is to allow the programmer to include sets of allowed letters? It could default to [A-Za-z] but maybe we could give them other options or the option to include an array of chars that are allowed? That might be nice if the text field is custom in a way that only allows certain specific characters.

Reply:
I like that idea!